### ISSUE TITLE
WebGL (WebGLPointsLayer) with transparent icons - how to extend hit area with semi-transparent background

### ISSUE THREAD

**Issue by messyjoes1** [openlayers/openlayers#15133]
I have icons which are mostly transparent but i want the whole icon area to be clickable/hoverable.

Working solution i currently have is drawing with canvas a near-transparent fill over icon 'rgba(0, 0, 0, 0.05)'. That works, but **fill is quite visible**, especially when icons overlap.

With regular VectorLayer i can get fill opacity down to 0.005 and still have pixel hit detection, but with WebGL the minimum opacity is 0.05. is there any better way to do hit detecion or achieve more transparent fill with WebGL ?

**Comment by jahow** [comment]
That's right, the cutoff value is defined here (for points): https://github.com/openlayers/openlayers/blob/70464b4f14a98b5fa5f7b6fe0367d259b0dff44c/src/ol/webgl/ShaderBuilder.js#L559

Since we're trying to replicate the general behavior of the canvas renderer, I think it would make sense to lower this value.

**Comment by mike-000** [comment]
> With regular VectorLayer i can get fill opacity down to 0.005

When using 2d canvas directly the supplied alpha is rounded to the nearest multiple of 1/255, so anything greater than 1/510 (or greater than or equal to 0.002) should be detectable even if not noticeable.  `ol/color#toString` is converting some colors with detectable alpha to transparent by rounding to the nearest 0.01.

**Comment by drew08t** [comment]
Is this something we can update? I'd be happy to create a PR if that is something you would be open to.

**Comment by jahow** [comment]
You are definitely welcome to create a PR for this! I think simply lowering the cutoff value is OK. This should be done for all shapes (lines, polygons and points).
